core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -108,22 +109,33 @@ abstract class AbstractIndex(@volatile private var _file: File, val baseOffset:
   @volatile
   protected var mmap: MappedByteBuffer = {
     val newlyCreated = file.createNewFile()
-    val raf = if (writable) new RandomAccessFile(file, "rw") else new RandomAccessFile(file, "r")
-    try {
-      /* pre-allocate the file if necessary */
-      if(newlyCreated) {
-        if(maxIndexSize < entrySize)
-          throw new IllegalArgumentException("Invalid max index size: " + maxIndexSize)
+    /* pre-allocate the file if necessary */
+    if(newlyCreated) {
+      if(maxIndexSize < entrySize)
+        throw new IllegalArgumentException("Invalid max index size: " + maxIndexSize)
+      val raf = if (writable) new RandomAccessFile(file, "rw") else new RandomAccessFile(file, "r")
```

Comment:
You can replace RandomAccessFile here with FileChannel just as we did at the other place and hence, eliminating the need for opening the file twice.